### PR TITLE
sync with upstream

### PR DESCRIPTION
**Pull request by zirain**
fix: #5462

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -55,7 +55,6 @@ build --host_linkopt=-pthread
 
 # CI sanitizer configuration
 #
-build:clang-asan-ci --config=clang-asan
```
Comment (kyessenov) [inline .bazelrc:58]: That simply disables asan, which is not the right fix.

File: envoy.bazelrc [hunk 2]
```diff
@@ -103,7 +105,6 @@ build:clang-tidy --output_groups=report
 build:clang-tidy --build_tag_filters=-notidy
 
 # Basic ASAN/UBSAN that works for gcc
-build:asan --action_env=ENVOY_ASAN=1
```
Comment (zirain) [inline envoy.bazelrc:106]: seems cause by this line, let me give a try.

File: .bazelrc [hunk 3]
```diff
@@ -53,17 +53,15 @@ build --cxxopt -Wformat-security
 # Link pthread for flatbuffers
 build --host_linkopt=-pthread
 
+build:clang --host_action_env=CC=
+build:clang --host_action_env=CXX=
```
Comment (zirain) [inline .bazelrc:57]: not sure why need this, cc @kyessenov

Comment (kyessenov) [reply .bazelrc:57]: I see, so this is an environment pollution I think from Istio CI.

**Review by kyessenov — APPROVED** [review]

**Pull request merged**